hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/tools/federation/RouterAdmin.java

```diff
@@ -1035,6 +1054,65 @@ private boolean updateQuota(String mount, long nsQuota, long ssQuota)
         .updateMountTableEntry(updateRequest);
     return updateResponse.getStatus();
   }
+  
+  /**
+   * initViewFsToMountTable.
+   * @param clusterName The specified cluster to initialize.
+   * @return If the quota was updated.
+   * @throws IOException Error adding the mount point.
+   */
+  public boolean initViewFsToMountTable(String clusterName)
+      throws IOException {
+    // fs.viewfs.mounttable.ClusterX.link./data
+    final String mountTablePrefix =
+        Constants.CONFIG_VIEWFS_PREFIX + "." + clusterName + "." +
+            Constants.CONFIG_VIEWFS_LINK + "./";
+    Map<String, String> viewFsMap = getConf().getValByRegex(mountTablePrefix);
+    if (viewFsMap.isEmpty()) {
+      System.out.println("There is no ViewFs mapping to initialize.");
+      return true;
+    }
+    for (Entry<String, String> entry : viewFsMap.entrySet()) {
+      Path path = new Path(entry.getValue());
+      DestinationOrder order = DestinationOrder.HASH;
+      String[] mount = entry.getKey().split(
+              clusterName + "." + Constants.CONFIG_VIEWFS_LINK + ".");
+      if (mount.length < 2) {
+        System.out.println("Added Mount Point failed " + entry.getKey());
+        continue;
+      }
+      String[] nss = new String[]{path.toUri().getAuthority()};
+      boolean added = addMount(
+          mount[1], nss, path.toUri().getPath(), false,
+          false, order, getACLEntityFormHdfsPath(path));
+      if (added) {
+        System.out.println("added mount point " + mount[1]);
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Returns ACLEntity according to a HDFS pat.
+   * @param path A path of HDFS.
+   */
+  public ACLEntity getACLEntityFormHdfsPath(Path path){
+    String owner = null;
+    String group = null;
+    FsPermission mode = null;
+    try {
+      FileSystem fs = path.getFileSystem(getConf());
+      if (fs.exists(path)) {
+        FileStatus fileStatus = fs.getFileStatus(path);
+        owner = fileStatus.getOwner();
+        group = fileStatus.getGroup();
+        mode = fileStatus.getPermission();
+      }
+    } catch (IOException e) {
+      System.out.println("Exception encountered " + e);
```

Comment:
.err.?

Reply:
I think it fails to get FileStatus and can continue to use the default ACLEntity to add mapping.

Reply:
Yes, but it should go to System.err not System.out, right?

Reply:
Yes,Thank you for reminding me.